SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -113,15 +142,103 @@ bool FSpatialGDKEditor::GenerateSchema(bool bFullScan)
 	return bResult;
 }
 
-bool FSpatialGDKEditor::LoadPotentialAssets(TArray<TStrongObjectPtr<UObject>>& OutAssets)
+void FSpatialGDKEditor::GetWorldDependencies(UWorld* World, TSet<FAssetData>& OutAssets, TSet<FAssetData>* AssetsToIgnore)
+{
+	TQueue<FName> DepsToSearch;
+	DepsToSearch.Enqueue(*World->GetOutermost()->GetPathName());
+
+	if (World->GetWorldSettings()->DefaultGameMode != nullptr)
+	{
+		UE_LOG(LogSpatialGDKEditor, Verbose, TEXT("Adding GameModeOverride: %s"), *World->GetWorldSettings()->DefaultGameMode->GetOutermost()->GetPathName())
+			DepsToSearch.Enqueue(*World->GetWorldSettings()->DefaultGameMode->GetOutermost()->GetPathName());
+	}
+	else
+	{
+		// Add default game mode
+		for (FString GameMode : TArray<FString>{ TEXT("GlobalDefaultGameMode"), TEXT("GlobalDefaultServerGameMode") })
+		{
+			FString GameModePath;
+			GConfig->GetString(
+				TEXT("/Script/EngineSettings.GameMapsSettings"),
+				*GameMode,
+				GameModePath,
+				GEngineIni
+			);
+
+			if (!GameModePath.IsEmpty())
+			{
+				UE_LOG(LogSpatialGDKEditor, Verbose, TEXT("Adding %s %s"), *GameMode, *FSoftObjectPath(GameModePath).GetLongPackageName());
+				DepsToSearch.Enqueue(*FSoftObjectPath(GameModePath).GetLongPackageName());
+			}
+		}
+	}
+
+	UE_LOG(LogSpatialGDKEditor, Verbose, TEXT("Loading all deps for %s"), *World->GetOutermost()->GetPathName());
+
+	if (World->WorldComposition != nullptr)
+	{
+		for (auto& Tile : World->WorldComposition->GetTilesList())
+		{
+			UE_LOG(LogSpatialGDKEditor, Verbose, TEXT("Adding World Comp Tile %s to Deps"), *Tile.PackageName.ToString());
+			DepsToSearch.Enqueue(Tile.PackageName);
+		}
+	}
+
+	FAssetRegistryModule& AssetRegistryModule = FModuleManager::LoadModuleChecked<FAssetRegistryModule>("AssetRegistry");
+	IAssetRegistry& AssetRegistry = AssetRegistryModule.Get();
+
+	TSet<FName> SearchedPackages;
```

Comment:
Add a quick comment about what SearchedPackages is being used for, also we add `NextDep` to it meaning the name of dependency and package is a little confusing